.travis.yml

```diff
@@ -0,0 +1,38 @@
+language: python
+python:
+  - "2.7"
+before_install:
+  - sudo apt-get update -qq
+  - sudo apt-get install mpi-default-bin mpi-default-dev
+
+  # LAMMPS
+  #  - build lammps as shared library
+  #  - and install LAMMPS python wrapper
+  - git clone git://git.lammps.org/lammps-ro.git mylammps
+  # checkout a recent stable version (from 28 June 2014)
+  - cd mylammps/src && git checkout r12164
+  # make shared library for python
+  - make makeshlib && make -f Makefile.shlib ubuntu_simple
+  - cd ../python && sudo python install.py /usr/lib/ $VIRTUAL_ENV/lib/python2.7/site-packages/
+  # make lammps executable and add to path
+  - cd ../src
+  - make ubuntu_simple
+  - ln -s lmp_ubuntu_simple lammps
+  - export PATH=$TRAVIS_BUILD_DIR/mylammps/src/:$PATH
+  - cd $TRAVIS_BUILD_DIR
+
+  # use current master of simphony
+  - pip install git+git://github.com/simphony/simphony-common.git#egg=simphony
```

Comment:
I see why it makes sense to add keys to identify where the key would be meaningful (so for example, adding a pair potential object to each point in your example, would in most cases be useless...

>Another problem is perfomance issues. The DataContainer in a number of places iterates over all the CUBA keywords defined in the CUBA enum this iteration depends on the length of the CUBA enum. As a result there is a small but measurable impact of having unused keywords in the enum that is used by the DataContainer

I am not sure why we have to loop over all the CUBA keys? would it be better to loop over the existing CUBA in an object and only treat those, rather than looping over the predefined keys? is there a performace reason for this? 

Please note that our CUBA keys is a very small list now, but will be few hundreds later on, a part of which may be used in each simulation or model. And there is no way we can generally determine in advance which subset of keyworkds can be used together for certain models, as the models themselves (i.e., simulations) will change depending on the use case and physics/chemistry needed.

>Finally I think it would be confusing for a user to see that a SD data container accepts configuration keywords. One would wonder if they are actually necessary. I would prefer a design where we do not have to provide additional documentation that these keywords while allowed in the container are ignored

from a user point of view, the modeler is expected to know which keys  (properties for the user) belong to which component. I do not feel, again from a modeler point of view, that this needs to be strictly imposed by code. The CUBA are knowledge based keywords, and the first layer of knowledge comes from the user, knowing where it makes sense to use them. The second one is, e.g., within the wrapper, that should "know" for which keywords to look for based on others, e.g., if in the BC defines the PERIODIC_BOUNDARY_CONDITION, then the wrapper should check if a SIMULAION_BOX is defined in the SD. This consistency of knowledge, is in my opnion important.